dsc_lib/src/util.rs

```diff
@@ -24,6 +39,95 @@ impl Default for DscSettingValue {
     }
 }
 
+/// Only `activity` and `percent_complete` fields are mandatory for Progress messages
+#[derive(Default, Debug, Clone, Serialize)]
+pub struct Progress {
+    pub activity:  String,
+    #[serde(default, skip_serializing_if = "is_default")]
+    pub status: String,
+    pub percent_complete: u16,
+    #[serde(default, skip_serializing_if = "is_default")]
+    pub seconds_remaining: u64,
```

Comment:
Make as Option.  Even `percent_complete` is probably an Option

Reply:
Updated.